layers/+lang/reason/packages.el

```diff
@@ -0,0 +1,160 @@
+;;; packages.el --- reason layer packages file for Spacemacs.
+;;
+;; Copyright (c) 2012-2016 Sylvain Benner & Contributors
+;;
+;; Author: Fredrik Dyrkell
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(defconst reason-packages
+  '(company
+    utop
+    reason-mode
+    merlin
+    popwin
+    flycheck
+    flycheck-ocaml
+    evil-matchit
+    ))
+
+(defun reason/post-init-company ()
+  (when (configuration-layer/package-usedp 'merlin)
+    (spacemacs|add-company-backends
+      :backends merlin-company-backend
+      :modes reason-mode))
+  )
+
+(defun reason/init-reason-mode ()
+  (use-package reason-mode
+    :defer t
+    :mode ("\\.rei?\\'" . reason-mode)
+    :init
+    (progn
+      (add-hook 'reason-mode-hook (lambda ()
```

Comment:
sdwolfz said about this hook within a hook:
>Reformating on save should be controlled by a variable, disabled by default

Reply:
I've had a go at restructuring this a bit and turned it off by default, with a note in the README about how to turn it on with a layer variable.